src/main/java/com/limechain/rpc/methods/system/SystemRPCImpl.java

```diff
@@ -132,9 +145,19 @@ public void systemRemoveReservedPeer(String peerId) {
     /**
      * Returns the state of the syncing of the node.
      */
-    // TODO: Implement in Mx.
     public Map<String, Object> systemSyncState() {
-        return null;
+        long highestBlock = this.connectionManager
+                .getPeerIds()
+                .stream()
+                .map(this.connectionManager::getPeerInfo)
+                .mapToLong(peerInfo -> peerInfo.getBestBlock().longValue())
+                .max().orElse(0);
```

Comment:
`.longValue()` is not a safe function, it won't crash if the BigInteger exceeds max long value. Is keeping it as a BigInteger a problem?

Reply:
No, it works with BigInteger as well